prometheus.lua

```diff
@@ -627,14 +648,14 @@ function Prometheus:metric_data()
   -- Force a manual sync of counter local state (mostly to make tests work).
   self._counter:sync()
 
-  local keys = self.dict:get_keys(0)
+  local keys = self.key_index:list()
   -- Prometheus server expects buckets of a histogram to appear in increasing
   -- numerical order of their label values.
   table.sort(keys)
```

Comment:
While running the benchmark, this occasionally produces an error:
```
2020/05/02 16:52:00 [error] 10#10: *417 lua entry thread aborted: runtime error: attempt to compare nil with string
stack traceback:
coroutine 0:
        [C]: in function 'sort'
        /nginx-lua-prometheus/prometheus.lua:654: in function 'metric_data'
        /nginx-lua-prometheus/prometheus.lua:700: in function 'collect'
        content_by_lua(nginx.conf:71):2: in function <content_by_lua(nginx.conf:71):1>, client: 172.17.0.1, server: benchmark, request: "GET /metrics HTTP/1.1", host: "localhost:18003"
```

Which means:
1. we need to remove `nil` values from the list of keys before sorting it;
2. we need to understand how `nil` values appeared in the list of keys during benchmark (which only adds metrics, never removing them).

Reply:
This is weird, but I managed to catch this even in the integration tests. The oddest thing is, that I printed the contents of the table and there is no `nil` in it. Also, the integration tests still pass, even though there are some 500 responses, that should be probably fixed too.

Reply:
Oh, now I see... There is a missing index and `table.sort` only works on continuous tables. Now just to find out how it happened when there is no delete.

Reply:
The problem was that `list` function returned reference to `self.keys`. This table was then sorted in `metric_data`, messing up al the indices. Stupid error :frowning_face: Should be fixed in 9e4d40ac05e8c24624f9bde237670e7dde0e7014 by returning a copy.